app/models/front_end_builds/build.rb

```diff
@@ -48,6 +48,10 @@ def self.find_best(params = {})
       elsif params[:job]
         query[:job] = params[:job]
 
+      elsif params[:company]
+        company = FrontEndBuilds::Company.where( app_id: app.id, name: params[:company] ).first
```

Comment:
I considered doing a fancy join against `companies` in order to have one less query, but I decided this code was simpler in a way that is 100% worth it.